### ISSUE TITLE
Strange Case. App crashed and removed suddenly. (Mibox)

### ISSUE THREAD

**Issue by 1an6** [yuliskov/SmartYouTubeTV#235]
it happen in my mibox 3 since yesterday, all of sudden, and happen a couple of times since then.

While watching video on Smart Youtube, it crashed suddenly and jump into main screen, then the smart Youtube app just disappear (uninstalled). (reinstalled apk and it works, crashed again once in a while)

troubleshooting attempt:
- update to the latest stable release (not work)
- factory setting on Mibox (not work)

**Comment by yuliskov** [comment]
Possible fix for Mi TV
https://github.com/yuliskov/SmartYouTubeTV/releases/download/6.16.29/SYTV_Orig_v6.16.29_r.apk

**Comment by 1an6** [comment]
hi, just tested. its the same.

**Comment by yuliskov** [comment]
thx, I'll continue to search the solution

**Comment by 1an6** [comment]
Thks, i will try to get you log file if possible.

**Comment by yuliskov** [comment]
Log is not useful here. I think that privacy guard or something like that blocks app.
Better try this old version: https://github.com/yuliskov/SmartYouTubeTV/releases/download/6.16.11/SYTV_Orig_v6.16.11_r.apk

**Comment by BigLeGorLee** [comment]
6.16.11 has been tested and also the same. Actually, this happens just suddenly. I use 6.15.45 before and it works well in a long time, I notice this since the system has a kinda upgrade action, then this situation happens.

**Comment by yuliskov** [comment]
Guys, thank you so much for the help.

**Comment by DrimonMaster** [comment]
Hi, same problem in mi tv 4 and mi box mini. disappear =(

Will wait for solution. 

Thanks for you work!

**Comment by cashara** [comment]
+1
I've the same issue on my MiTV 4.
App started to dissapear two days ago.

**Comment by hzleochang** [comment]
It's the same problem of my Mibox3 pro and happened just 1 day ago,App will disappeared and no change even you reinstall it.
it happened in 6.16.29 and also 6.15.45...
Dose it banned by Mi system?
Hope you can fix it,thanks a lot!

**Comment by 1an6** [comment]
it seems like all happen on the same day. i am sharing my android info here. hope it helps for troubleshoot.
![img_0982](https://user-images.githubusercontent.com/29710203/52174818-b08b9b00-27d4-11e9-9df4-ec8f6b11a0ae.jpg)

**Comment by Cappuccin0** [comment]
The app has been removed twice on my MiBox 4 since yesterday, I noticed the one was after standby and wakeup and another one happened after system reboot.

I happen to have a few apks for older builds, and so far ver 6.12.58 works perfectly for me (tested with a couple of reboots).

**Comment by tourmalinec** [comment]
exactly the same issue here.  The app worked well before, but since some days ago it started to crash instead of a proper exit when a video ended.  Now it actually gets removed from the system when it crashes (and it will crash in the middle of playing or pausing video ).   It almost looks like that the system periodically scans and force-removes this app.  Tried to install different versions of Smart Youtube TV, no help.   I am using a MiBox 3S with Chinese system installed.  
Other manually installed apps, e.g. Yalp store, Es File explorer. google play store, Youtube, etc. don't have this issue.

**Comment by KonstantinBabenko** [comment]
Same issue here for Xiaomi Mi TV3. Started 2 days ago. App just gets deleted periodically.
P.S. Also some videos from my subscriptions started lagging.

**Comment by philipwai99** [comment]
Same issue here for XiaoMi Mi TV4A. It happened on feb 2 night Hong Kong Time. App crashed and uninstalled suddenly. I retried some versions today but the issue repeated.

**Comment by yuliskov** [comment]
Guys, another try:
https://github.com/yuliskov/SmartYouTubeTV/releases/download/6.16.31/SYTV_Orig_v6.16.31_r.apk

**Comment by KonstantinBabenko** [comment]
@yuliskov This version fails to start on my MiTV3

**Comment by wbaixz** [comment]
@yuliskov This version fails to start on my MiTV4

**Comment by marleysaint** [comment]
This version fails to start on my MiTV4S

**Comment by yuliskov** [comment]
update:
https://github.com/yuliskov/SmartYouTubeTV/releases/download/6.16.32/SYTV_Orig_v6.16.32_r.apk

**Comment by marleysaint** [comment]
> update:
> https://github.com/yuliskov/SmartYouTubeTV/releases/download/6.16.32/SYTV_Orig_v6.16.32_r.apk
This version fails to start on my MiTV4S

**Comment by marleysaint** [comment]
rather start, but left after 5 minutes

**Comment by yuliskov** [comment]
OK. Another try.
https://github.com/yuliskov/SmartYouTubeTV/releases/download/6.16.33/SYTV_Orig_v6.16.33_r.apk

**Comment by davidghuang** [comment]
Had the same problem, tried the latest 6.16.33_r version. So far so good. Looks like this one will stay. Just out of curiosity, is the "bug" caused by Mi box?

**Comment by yuliskov** [comment]
> is the "bug" caused by Mi box?

Definitely, yes. App is banned by privacy guard or something like this. What could cause such behavior I don't know.

**Comment by davidghuang** [comment]
It's not nice for a tv box to uninstall a user's app without even a warning. Maybe it's time for me to switch to a different one.

**Comment by yuliskov** [comment]
@davidghuang lets hope that problem won't repeat in the future

**Comment by 1an6** [comment]
hi all, just a quick update on my "temporary" solution.
somehow i was able to launch the app through a third party tool named Shafa, and it works for hours without crashing for now.

i don think version really matter here, as it was freaking xiaomi removed our apps (or concealed it from the desktop) i believe. 

give it a shot and let us know if you find out anything, and hopefully we can solve this permanently soon

**Comment by yuliskov** [comment]
@1an6 thx, men. Appreciate your help.

**Comment by yuliskov** [comment]
@1an6 I wanna know, this happen only for my app or other apps removed too?

**Comment by 1an6** [comment]
> @1an6 I wanna know, this happen only for my app or other apps removed too?

its seems only happen to your app bro. my vlc player works fine.

**Comment by terranzmczmx** [comment]
@yuliskov I got the same problem.....Help!! T_T

**Comment by tourmalinec** [comment]
> OK. Another try.
> https://github.com/yuliskov/SmartYouTubeTV/releases/download/6.16.33/SYTV_Orig_v6.16.33_r.apk

So far this version works well for me

**Comment by 1an6** [comment]
> It's not nice for a tv box to uninstall a user's app without even a warning. Maybe it's time for me to switch to a different one.

yes, bug solved or not, i swear this will be my last purchase of mibox. (angry)

**Comment by benjaminv** [comment]
> > It's not nice for a tv box to uninstall a user's app without even a warning. Maybe it's time for me to switch to a different one.
> 
> yes, bug solved or not, i swear this will be my last purchase of mibox. (angry)

Agreed disgusting behaviour.

**Comment by yuliskov** [comment]
@terranzmczmx what about this link https://github.com/yuliskov/SmartYouTubeTV/issues/235#issuecomment-460085161

**Comment by cloudsupll** [comment]
SYTV_Orig_v6.16.33_r.apk mibox3无法运行

**Comment by philipwai99** [comment]
My MiTV 4a works fine with the 33r release. My kids watched half-day of youtube with no crash.
Thanks a lot!

**Comment by KonstantinBabenko** [comment]
@yuliskov App does not get deleted today with the version you posted. But some (not all) videos are lagging now. (for example this one https://www.youtube.com/watch?v=dZaDby37Ut8&t=1s) There were no lags on 6.15.
PS I also wonder can you map OK and arrow buttons for Login to youtube screen? Because when app gets removed and I installed it a several times during past days it was really frustrating to connect a mouse to click Login button.
PS After videos started to lag I switched from 4kPro launcher to 4k alternative and videos continued to lag there. When I switched back SmartYoutubeTv logged out me for some reason. I had to connect it again. This might be a bug, right?

**Comment by yuliskov** [comment]
@KonstantinBabenko login bug should be fixed here. Could you confirm that?
https://github.com/yuliskov/SmartYouTubeTV/releases/download/6.16.36/SYTV_Orig_v6.16.36_r.apk

**Comment by marleysaint** [comment]
> @KonstantinBabenko login bug should be fixed here. Could you confirm that?
> https://github.com/yuliskov/SmartYouTubeTV/releases/download/6.16.36/SYTV_Orig_v6.16.36_r.apk

Yes

**Comment by yuliskov** [comment]
@marleysaint thx

**Comment by LegoCodr** [comment]
I've encoutered the same problem, my MiTv 4a uninstalled my SmartYoutubeTV app like a hundred times. I tried so many different versions, and it keeps doing that. Shame on xiaomi!!!

**Comment by yuliskov** [comment]
@quiary are you sure that problem still appears on this version
https://github.com/yuliskov/SmartYouTubeTV/releases/download/6.16.36/SYTV_Orig_v6.16.36_r.apk

**Comment by duoxoud** [comment]
Tried https://github.com/yuliskov/SmartYouTubeTV/releases/download/6.16.36/SYTV_Orig_v6.16.36_r.apk
So far so good

**Comment by KonstantinBabenko** [comment]
@yuliskov Installed 6.16.40. Issue with login seems to be resolved. OK button seem to work there as well. Thank you.
Video lagging is still an issue though. The sound is good during lags. Also I had a several videos from my subscription feed that started with black screen and sound going. When I changed video codec from avc to vp9 or vice versa then black screen is gone, but video still lags.

**Comment by terranzmczmx** [comment]
> @terranzmczmx what about this link [#235 (comment)](https://github.com/yuliskov/SmartYouTubeTV/issues/235#issuecomment-460085161)

Yes, it works well. Thx.

**Comment by sergruss** [comment]
Installed 6.16.40 8 h ago. It still works on Mi TV 4S.

But if open a context menu during viewing of video, it disappears after a while in independence of that how long there was a movement according to the menu.

**Comment by philipwai99** [comment]
6.16.40 works fine on my MiTV 4A.
Thanks for your great work.

**Comment by matchsun** [comment]
I met same issue with my mi TV, I will not buy any products from xiaomi next time.

**Comment by LegoCodr** [comment]
The latest r42 version really solves all the problems I had, thanks. Mi tv no longer detele my app, and all the functions are working fine for me. Been using this ver for two days now. Good job!

**Comment by LegoCodr** [comment]
By the way, r33, r36, r40 can all survive on my MiTV 4a, but can not Auto play the next video, just show loading and then a black screen. But the r42 works really great.

**Comment by busyleung** [comment]
> Possible fix for Mi TV
> https://github.com/yuliskov/SmartYouTubeTV/releases/download/6.16.29/SYTV_Orig_v6.16.29_r.apk

App keeps disappearing after install these days on my Xiaomi TV 3S. Xiaomi really block smart youtube tv app.

**Comment by yuliskov** [comment]
Bug is fixed in the latest version.

**Comment by Justsoos** [comment]
most like the behavior of censorship, they (Chinese GOV) break any app of youtube remotely or by schedule. 
you have to set up a non-original android system, or re-package the app with different name to avoid censorship.

could SYouTube owner give us a special version which random app's name like magisk?

BTW, you guys Russian DO NOT fear of the censorship even turning off Internet from Putin??? :p

**Comment by laye0619** [comment]
Hey yuliskov, are you still working on giving us a special version which random app name? I do believe it can solve the ban problem.

**Comment by yuliskov** [comment]
@laye0619 no, I don't. I think, this is not worth such efforts.

**Comment by laye0619** [comment]
> @laye0619 no, I don't. I think, this is not worth such efforts.

Got it. It seems that your latest version works fine on Mi device, right?
